src/DataWrangling/ECCO/ECCO_restoring.jl

```diff
@@ -231,52 +234,65 @@ end
+                  time_indexing = Cyclical(),
+                  mask = 1,
+                  rate = 1 / 20days,
+                  grid = nothing,
+                  maxiter = Inf)
 
 Create a restoring forcing term that restores to values stored in an ECCO field time series.
 
 # Arguments:
 =============
-- `metadata`: The metadata for the ECCO field time series.
+- `variable_name`: The name of the variable to restore. The choice is between 
+                   :temperature, :salinity, :u_velocity, :v_velocity, :sea_ice_thickness, :sea_ice_area_fraction.
+- `architecture`: The architecture. Typically `CPU` or `GPU`. Default is `CPU`.
 
 # Keyword Arguments:
 ====================
-- `architecture`: The architecture. Typically `CPU` or `GPU`
+- `version`: The version of the ECCO dataset. Default is `ECCO4Monthly()`.
+- `dates`: The dates to use for the ECCO dataset. Default is `all_ECCO_dates(version)`.
 - `time_indices_in_memory`: The number of time indices to keep in memory. trade-off between performance
                             and memory footprint.    
-- `time_indexing`: The time indexing scheme for the field time series, see [`FieldTimeSeries`](@ref)
+- `time_indexing`: The time indexing scheme for the field time series≥
 - `mask`: The mask value. Can be a function of `(x, y, z, time)`, an array or a number
-- `timescale`: The restoring timescale.
+- `rate`: The restoring rate in s⁻¹.
+- `time_indices_in_memory = 2, # Not more than this if we want to use GPU!
+- `maxiter`: maximum number of iterations for the inpainting algorithm. (defaults to `Inf`)
+
+It is possible to also pass an `ECCOMetadata` type as the first argument without the need for the 
+`variable_name` argument and the `version` and `dates` keyword arguments.
 """
-function ECCO_restoring_forcing(variable_name::Symbol, version=ECCO4Monthly(); kw...) 
-     metadata = ECCOMetadata(variable_name, all_ECCO_dates(version), version)
-    return ECCO_restoring_forcing(metadata; kw...)
+function ECCORestoring(variable_name::Symbol, architecture = CPU(); 
+                       version = ECCO4Monthly(),
+                       dates = all_ECCO_dates(version), 
+                       kw...) 
+                                
+     metadata = ECCOMetadata(variable_name, dates, version)
+    return ECCORestoring(metadata; architecture, kw...)
 end
 
-function ECCO_restoring_forcing(metadata::ECCOMetadata;
-                                architecture = CPU(), 
-                                time_indices_in_memory = 2, # Not more than this if we want to use GPU!
-                                time_indexing = Cyclical(),
-                                mask = 1,
-                                timescale = 20days,
-                                grid = nothing)
+function ECCORestoring(metadata::ECCOMetadata;
+                       architecture = CPU(), 
+                       time_indices_in_memory = 2, # Not more than this if we want to use GPU!
```

Comment:
should this be `backend`?

Reply:
At the moment, the only supported backend is the `ECCOBackendNetCDF`. If we want to be able to preprocess data we can change this to accept a backend and support different backends.